lms/djangoapps/instructor/views/api.py

```diff
@@ -1228,6 +1228,29 @@ def get_enrollment_report(request, course_id):
         })
 
 
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+@require_finance_admin
+def get_exec_summary_report(request, course_id):
+    """
+    get the executive summary report for the particular course.
+    """
+    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    try:
+        instructor_task.api.submit_executive_summary_report(request, course_key)
+        status_response = _("Your Executive Summary report is being generated! "
+                            "You can view the status of the generation task in"
+                            " the 'Pending Instructor Tasks' section.")
```

Comment:
Your executive summary report is being created.
To view the status of the report, see the "Pending Instructor Tasks" section.